x/core/keeper/gas_meter.go

```diff
@@ -63,9 +62,82 @@ func (k Keeper) DistributionsFromGasMeter(ctx sdk.Context, reqID string, reqHeig
 	return dists
 }
 
+func (k Keeper) ProcessDistributions(ctx sdk.Context, dists []types.Distribution, dr *types.DataRequest, minimumStake math.Int) error {
```

Comment:
Considering in `endblock.go` in the same module we call `GetGasMeterResults` and then immediately call this function- it would seem we can just merge them? I.e. no need to build the distribution list in the first place. Or do we ever plan to have logic in-between those two steps?